pkgs/development/compilers/javacard-devkit/default.nix

```diff
@@ -0,0 +1,86 @@
+{ stdenv, requireFile, unzip, makeWrapper, oraclejdk8 }:
+
+let
+  pkg = "javacard-devkit";
+  version = "2.2.2";
+
+  underscoreVersion = builtins.replaceStrings ["."] ["_"] version;
+
+  downloadUrl = "http://www.oracle.com/technetwork/java/javasebusiness/downloads/java-archive-downloads-javame-419430.html#java_card_kit-2.2.2-oth-JPR";
+in
+
+stdenv.mkDerivation rec {
+  name = "${pkg}-${version}";
+
+  src = requireFile {
+    name = "java_card_kit-${underscoreVersion}-linux.zip";
+    url = downloadUrl;
+    sha256 = "1rzkw8izqq73ifvyp937wnjjc40a40drc4zsm0l1s6jyv3d7agb2";
+  };
+
+  nativeBuildInputs = [ unzip makeWrapper ];
+
+  unpackPhase = "true";
+
+  buildPhase = ''
+    unzip $src
+
+    cd "java_card_kit-${underscoreVersion}"
+    unzip "java_card_kit-${underscoreVersion}-rr-bin-linux-do.zip"
+
+    chmod +x bin/{apdutool,capdump,capgen,converter,cref,exp2text,jcwde,scriptgen,verifycap,verifyexp,verifyrev}
+    chmod u+w bin/cref
+    patchelf --set-interpreter "$(cat "$NIX_CC/nix-support/dynamic-linker")" bin/cref
+
+    mkdir "$out"
+    cp -R bin "$out"
+    cp -R lib "$out"
+    cp -R api_export_files "$out"
+
+    for p in "$out/bin/"{apdutool,capdump,capgen,converter,cref,exp2text,jcwde,scriptgen,verifycap,verifyexp,verifyrev}; do
+      wrapProgram "$p" --set JAVA_HOME "${oraclejdk8}" --prefix CLASSPATH : "$out/api_export_files"
+    done
+
+    makeWrapper "${oraclejdk8}/bin/javac" $out/bin/javac --prefix CLASSPATH : "$out/lib/api.jar"
+    # TODO (question for the reviewer): The resulting javac can then be used with:
+    #     javac -source 1.2 -target 1.2 myPackage/myApplet.java
+    # Should I include the -source/-target arguments in this wrapper?
+    # I'm thinking this may break unaware users just adding javacard-devkit to their env
```

Comment:
If it's in the same environment as the JDK, it will collide with the `javac` from the JDK, but maybe it's a good idea to use a different name?

Reply:
Hmm, the issue is all the documentation online mentions only using `javac` for compiling (well, when there is any kind of documentation, that is), so I fear if we call it `javacardc` people will be lost… but maybe that's the least bad option? Having `javacardc` that would have both the right `CLASSPATH` and the right `-source 1.5 -target 1.5`…

Then, all tutorials online tell you to set the `CLASSPATH` and `-source 1.5 -target 1.5` by hand, so I wonder how far we should go from upstream for end-user convenience… I think for `CLASSPATH` given it's hidden somewhere in the store there's little room for argumentation, but for `-source`/`-target` I wonder.